### PR TITLE
[T4A7][W11-B4]Ong Heng Le 

### PR DESCRIPTION
**Pull request by initialshl**

### CODE REVIEW

**Comment by initialshl** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Name.java [hunk 1]
```diff
@@ -43,6 +46,48 @@ public static boolean isValidName(String test) {
...
+    public boolean isSimilar(Name other) {
+        if (other == null)
+            return false;
+        
+        List<String> myWords = new ArrayList<String>(getWordsInName());
+        List<String> otherWords = new ArrayList<String>(other.getWordsInName());
+        
+        // make both lists of words into lowercase
+        ListIterator<String> iterator = myWords.listIterator();
+        while (iterator.hasNext())
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Name.java]: coding standard violation :-1:  the brace should be immediately after `while` statement not next line

File: src/seedu/addressbook/data/person/Name.java [hunk 2]
```diff
@@ -43,6 +46,48 @@ public static boolean isValidName(String test) {
...
         return Arrays.asList(fullName.split("\\s+"));
     }
+    
+    /**
+     * Returns true of the other name is very similar to this name. 
+     * Two names are considered similar if they have at least 
+     * 50 percent match in case-insensitive words, in any order.
+     */
+    public boolean isSimilar(Name other) {
+        if (other == null)
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Name.java]: coding standard violation.  no braces around `if`

File: test/java/seedu/addressbook/data/person/NameTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    @Before
+    public void setup() throws IllegalValueException {
+        name = new Name("John Doe");
+    }
+    
+    @Test
+    public void testIsSimilar() throws IllegalValueException {
+        
+        // null input
+        assertFalse(name.isSimilar(null));
```
Comment (okkhoy) [inline test/java/seedu/addressbook/data/person/NameTest.java]: it is good to see you have considered quite a few test cases. :+1:

**Comment by okkhoy** [comment]
@initialshl 
Looking at your commits, I can see that the order of test-develop is little messed up. I hope the tutorial discussions give an idea on how to start off with TDD.

Some comments added. Please ack & close the PR after reading comments.

**Pull request closed without merging**